src/server/worker/master.go

```diff
@@ -78,10 +80,7 @@ func (a *APIServer) master() {
 	// to restart.
 	b.InitialInterval = 10 * time.Second
 	backoff.RetryNotify(func() error {
-		ctx, cancel := context.WithCancel(a.pachClient.AddMetadata(context.Background()))
-		defer cancel()
-
-		ctx, err := masterLock.Lock(ctx)
+		ctx, err := masterLock.Lock(a.pachClient.AddMetadata(context.Background()))
```

Comment:
Looks like we are getting rid of a `cancel()` here.  Is that intended?

Reply:
It is, that `cancel` was only ever called by the `defer cancel()` below it so it wasn't serving any purpose that I could tell.

Reply:
Not sure if I understand.  The point is that if we ever retry in this retry loop, we cancel the context so that all network calls that were triggered in the previous retry iteration are cancelled.